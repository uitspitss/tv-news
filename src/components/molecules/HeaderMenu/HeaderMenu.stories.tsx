import React from 'react';
import { Story, Meta } from '@storybook/react/types-6-0';
import base from 'paths.macro';

import { HeaderMenu, HeaderMenuProps } from './HeaderMenu';

const path = base.replace(
  /^.+\/(atoms|molecules|organisms|templates|pages)\/(.+)\//,
  '$1/$2',
);

export default {
  title: `${path}`,
  component: HeaderMenu,
} as Meta;

const Template: Story<HeaderMenuProps> = (args) => <HeaderMenu {...args} />;

export const Desktop = Template.bind({});

export const Mobile = Template.bind({});
Mobile.parameters = {
  viewport: {
    defaultViewport: 'iphonex',
  },
};
