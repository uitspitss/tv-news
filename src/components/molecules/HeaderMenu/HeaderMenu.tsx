import React, { memo, VFC } from 'react';
import tw, { styled } from 'twin.macro';
import Link from 'next/link';

export type HeaderMenuProps = {
  /**
   * class for twin.macro
   */
  className?: string;
};

type Props = HeaderMenuProps;

const Component: VFC<Props> = (props) => {
  const { className } = props;

  return (
    <div className={className}>
      <Link href="/users-ssr">
        <a>Users SSR</a>
      </Link>
      <Link href="/users-ssg">
        <a>Users SSG</a>
      </Link>
      <Link href="/users-swr">
        <a>Users useSWR</a>
      </Link>
      <Link href="/users-recoil">
        <a>Users useUsers</a>
      </Link>
    </div>
  );
};

const StyledComponent = styled(Component)`
  ${tw`text-sm lg:flex-grow`}

  & > a {
    ${tw`block mt-4 lg:inline-block lg:mt-0 text-gray-500`}

    &:not(:last-child) {
      ${tw`mr-4`}
    }
  }
`;

/**
 * menu in header
 */
export const HeaderMenu = memo((props: HeaderMenuProps) => {
  return <StyledComponent {...props} />;
});
