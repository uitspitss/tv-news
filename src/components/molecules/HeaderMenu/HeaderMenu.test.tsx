import React from 'react';
import { render } from '@testing-library/react';

import { HeaderMenu } from './HeaderMenu';

describe('HeaderMenuList', () => {
  test('snapshot test', () => {
    const { container } = render(<HeaderMenu />);
    expect(container).toMatchSnapshot();
  });
});
