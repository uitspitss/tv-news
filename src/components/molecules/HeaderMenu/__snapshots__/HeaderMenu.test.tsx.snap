// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderMenuList snapshot test 1`] = `
.emotion-0 {
  font-size: 0.875rem;
  line-height: 1.25rem;
}

@media (min-width: 1024px) {
  .emotion-0 {
    -webkit-box-flex: 1;
    -webkit-flex-grow: 1;
    -ms-flex-positive: 1;
    flex-grow: 1;
  }
}

.emotion-0>a {
  display: block;
  margin-top: 1rem;
  --tw-text-opacity: 1;
  color: rgba(107, 114, 128, var(--tw-text-opacity));
}

@media (min-width: 1024px) {
  .emotion-0>a {
    display: inline-block;
    margin-top: 0px;
  }
}

.emotion-0>a:not(:last-child) {
  margin-right: 1rem;
}

<div>
  <div
    class="emotion-0 emotion-1"
  >
    <a
      href="/users-ssr"
    >
      Users SSR
    </a>
    <a
      href="/users-ssg"
    >
      Users SSG
    </a>
    <a
      href="/users-swr"
    >
      Users useSWR
    </a>
    <a
      href="/users-recoil"
    >
      Users useUsers
    </a>
  </div>
</div>
`;
